tests/unittest/_torch/visual_gen/test_wan_i2v_teacache.py

```
@@ -0,0 +1,236 @@
+    def test_wan22_raises_if_teacache_enabled(self):
+        if not os.path.exists(WAN22_I2V_A14B_PATH):
+            pytest.skip(
+                f"Checkpoint not found: {WAN22_I2V_A14B_PATH} (set DIFFUSION_MODEL_PATH_WAN22_I2V)"
+            )
+        args = DiffusionArgs(
+            checkpoint_path=WAN22_I2V_A14B_PATH,
+            device="cuda",
+            dtype="bfloat16",
+            teacache=TeaCacheConfig(enable_teacache=True),
+        )
+        with pytest.raises(ValueError, match="TeaCache is not supported for Wan 2.2"):
```

Ruff (0.15.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)